### ISSUE TITLE
Accessibility: Buy with Google Pay Button missing focus outline

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43622]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The "buy with Google Pay" button doesn't have a focus outline. There is a style declaration in Stripe's stylesheet removing outlines on focus:
<img width="1200" alt="Screenshot of the Google Pay button and related CSS in the browser dev tools." src="https://github.com/woocommerce/woocommerce/assets/13153456/6ff1bbd5-de0b-4a90-84ca-c0b1be44d229">

```
:focus {
    outline: none;
}
```

I'm not certain if you're able to resolve this in WooCommerce core because it's an iframe, but I'm assuming you have more sway at Stripe than I do.

### Expected behavior

All focusable elements should receive an outline on focus. Color alone should not be used to indicate the state.

### Actual behavior

There is not a 

### Steps to reproduce

On the cart page in Google Chrome, tab to the "Buy with Google Pay" button. 

### WordPress Environment

N/A

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by frosso** [comment]
Hi @amberhinds , thank you for the report!

As you mentioned, WooCommerce can't change the outline styles of the button because Stripe renders it within an `iframe` element that we don't control.

> All focusable elements should receive an outline on focus. Color alone should not be used to indicate the state.

I'd like to clarify on this - [the "focus visible" 2.4.7 success criterion of the WCAG 2.1 guidelines](https://www.w3.org/TR/WCAG21/#focus-visible), doesn't explicitly state that an outline is needed. It states that the indicator must be visible.

The Google Pay button (provided in the example) already changes color when focus is received.
![2024-06-05 11 17 40](https://github.com/woocommerce/woocommerce/assets/273592/f89b662c-393a-4a34-b273-91a1e7dcf49c)

Could you please expand on why you would expect this button to also receive an outline?

Thank you again for your time!

**Comment by amberhinds** [comment]
The relevant success criteria is [1.4.1 Use of Color (Level A)](https://www.w3.org/TR/WCAG21/#use-of-color) which states:
> Color is not used as the only visual means of conveying information, indicating an action, prompting a response, or distinguishing a visual element.

That success criterion is the one that requires there to be some other indicator. It doesn't technically have to be an outline, there could be some other indicator that shows the change of state (such as the presence of an icon on the button only on focus or going from solid to outline style, etc.). It's just that the typical and expected indicator is an outline, so that's what I would recommend.

It's also worth noting that the color of the button above (when I inspect it) is `#353739` - a dark gray, which is barely different from black that the button is unfocused. It has a [1.75:1 contrast ratio](https://webaim.org/resources/contrastchecker/?fcolor=353739&bcolor=000000) between the focus/hover and default states. For people with low vision, this level of contrast may be completely indistinguishable.

**Comment by frosso** [comment]
Thank you for the additional information @amberhinds ! 

I already communicated about the insufficient contrast ratio to the Stripe team, I'll also convey the need for an additional indicator besides the color change 👍 

I'll keep relaying updates here as I receive them.

For internal reference: p1717662431926369/1717603559.231909-slack-C9976E5MJ

**Comment by amberhinds** [comment]
Thanks!

**Comment by frosso** [comment]
I have an update!

> the contrast ratio is coming from Google.
> You can see / try this out in their docs: https://developers.google.com/pay/api/web/guides/brand-guidelines
> For the focus indicator, the Google Pay button in ECE does have an outline, but not in PRB.

[The new version of WooPayments](https://github.com/Automattic/woocommerce-payments/releases/tag/8.0.0) coincidentally uses ECE to render the GooglePay/ApplePay buttons.

I tested this behavior, and I see the focus outline on the "light" and "outline" variants of the button, but it's not visible or distinguishable on the dark one.

![2024-08-01 10 16 39](https://github.com/user-attachments/assets/4c71cf25-7e8b-41a5-8580-d7f36c2194ba)
![2024-08-01 10 16 27](https://github.com/user-attachments/assets/6d04db88-f70d-42a4-ae3b-5a08ef76ccff)
![2024-08-01 10 16 14](https://github.com/user-attachments/assets/9d75f399-b2f8-452d-832e-cf2b81842bb2)

I wanted to let you know that I reached out to Stripe again for further clarification.

**Comment by amesplant** [comment]
@frosso are we in a holding pattern on this until we here back from Stripe?

**Comment by frosso** [comment]
@amesplant correct, it's in Stripe's court at this time

**Comment by vikrampm1** [comment]
Hi @frosso is there an update from Stripe?

**Comment by frosso** [comment]
@vikrampm1 not yet, I solicited it.

**Comment by jeffpaul** [comment]
@slash1andy @johndcoy do either of you know who the partner manager might be with Stripe to try and help nudge this one along (as it will help close/unblock a Woo core accessibility issue)?

**Comment by vikrampm1** [comment]
@frosso @slash1andy @johndcoy following up on the above if there is an update from Stripe.

**Comment by frosso** [comment]
It looks like the outline has been fixed in the last few months by Stripe

![Image](https://github.com/user-attachments/assets/ec41dc16-fd0f-4574-82e0-3c3602e492df)
![Image](https://github.com/user-attachments/assets/cea957de-9f34-457b-882c-00ad5202f47a)

I followed up and received this message:

> we are following the style guideline provided by Google which you can see here: https://developers.google.com/pay/api/web/guides/brand-guidelines. We won't be able to make changes that don't align with their best practices.

I confirmed that the button in question implemented within WooPayments (which is [the Stripe ECE button](https://docs.stripe.com/elements/express-checkout-element)) behaves similarly to the GooglePay guidelines.
Since the focus outline is now visible and is adhering to the guidelines, we can consider this closed.
